crates/wasm-smith/src/code_builder.rs

```diff
@@ -87,6 +91,8 @@ instructions! {
     (Some(return_valid), r#return, 900),
     (Some(call_valid), call),
     (Some(call_indirect_valid), call_indirect),
+    (Some(throw_valid), throw),
```

Comment:
I might be worth putting a weight on this one like `nop` and the others since this is probably going to get generated a lot otherwise and it's not too too interesting if modules mostly just immediately `throw`

Reply:
Good idea. Thinking on this more, it feels like there are two distinct cases here:
  1. throw with no enclosing catch (acts like an unreachable, basically a panic)
  2. throw to an enclosing catch that can handle it (acts like a br)

It would be nice to generate (2) more frequently than (1) so that we can test exceptional control flow well. Is there a way to make weight conditional on context? Or would that have to be achieved by splitting throw_valid/throw into two different instruction emitters, for the two different cases?

Reply:
Yeah for that I think it can be done with separate emitters. We haven't had that come up yet so it's breaking new ground!

Reply:
This is actually a bit challenging, so I'm going to defer it. The problem is that throws in a try block are generated before the catch blocks that describe which tags can be caught in that try. It's solvable, but I think keeping it simple is best to start.